src/common.cpp

```diff
@@ -1993,3 +1994,26 @@ void redirect_tty_output() {
     show_stackframe(L'E', 99, 1);
     abort();
 }
+
+/// Return a path to a directory where we can store temporary files.
+wcstring get_path_to_tmp_dir()
+{
+    env_var_t env_tmpdir = env_get_string(L"TMPDIR");
+    if (!env_tmpdir.missing()) return env_tmpdir;
+
+#ifdef _CS_DARWIN_USER_TEMP_DIR
+    char osx_tmpdir[31];
+    confstr(_CS_DARWIN_USER_TEMP_DIR, osx_tmpdir, 30);
```

Comment:
This can fail. You need to check whether or not it returns zero. You also need to check if the value returned exceeds PATH_MAX which indicates the value was truncated and therefore should not be used.